dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/SerializerTest.java

```diff
@@ -0,0 +1,643 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.dynatrace.metric.util.Dimension;
+import com.dynatrace.metric.util.DimensionList;
+import com.dynatrace.metric.util.MetricBuilderFactory;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.data.*;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.*;
+import org.junit.jupiter.api.Test;
+
+class SerializerTest {
+  private static final Serializer serializer =
+      new Serializer(MetricBuilderFactory.builder().build());
+
+  @Test
+  void fromLabels() {
+    Labels labels = Labels.of("label1", "value1", "label2", "value2");
+    DimensionList expected =
+        DimensionList.create(
+            Dimension.create("label1", "value1"), Dimension.create("label2", "value2"));
+    DimensionList actual = Serializer.fromLabels(labels);
+
+    assertEquals(expected.getDimensions(), actual.getDimensions());
+    assertNotSame(expected, actual);
+  }
+
+  @Test
+  void fromLabelsEmpty() {
+    Labels labels = Labels.empty();
+    DimensionList expected = DimensionList.create();
+    DimensionList actual = Serializer.fromLabels(labels);
+
+    assertEquals(expected.getDimensions(), actual.getDimensions());
+    assertNotSame(expected, actual);
+  }
+
+  @Test
+  void fromLabelsNormalize() {
+    Labels labels = Labels.of("~~!123", "test", "!!test2", "test2");
+    DimensionList expected = DimensionList.create(Dimension.create("test2", "test2"));
+    DimensionList actual = Serializer.fromLabels(labels);
+
+    assertEquals(expected.getDimensions(), actual.getDimensions());
+    assertNotSame(expected, actual);
+  }
+
+  @Test
+  void createLongSumLinesCumulative() {
+    Collection<LongPointData> longPointDataCollection =
+        new ArrayList<LongPointData>() {
+          {
+            add(
+                LongPointData.create(
+                    1619687639000000000L, 1619687659000000000L, Labels.empty(), 123L));
+            add(
+                LongPointData.create(
+                    1619687639000000000L, 1619687659000000000L, Labels.empty(), 321L));
+          }
+        };
+    LongSumData longSumData =
+        LongSumData.create(true, AggregationTemporality.CUMULATIVE, longPointDataCollection);
+    MetricData metricData =
+        MetricData.createLongSum(
+            Resource.getDefault(),
+            InstrumentationLibraryInfo.empty(),
+            "longSumData",
+            "",
+            "",
+            longSumData);
+
+    List<String> longSumLines = serializer.createLongSumLines(metricData, false);
+    assertThat(longSumLines).hasSize(2);
+    assertThat(longSumLines.get(0)).isEqualTo("longSumData count,123 1619687659000");
+    assertThat(longSumLines.get(1)).isEqualTo("longSumData count,321 1619687659000");
+  }
+
+  @Test
+  void createInvalidLongSumLinesCumulative() {
```

Comment:
What's invalid here?

Reply:
you're right, a zero timestamp is not invalid, it will just be discarded and serialized without timestamp. i've updated all the other tests as well